skiko/src/awtMain/objectiveC/macos/MetalContextHandler.mm

```diff
@@ -0,0 +1,64 @@
+#ifdef SK_METAL
+
+#import <jawt.h>
+#import <jawt_md.h>
+
+#import <QuartzCore/CAMetalLayer.h>
+#import <Metal/Metal.h>
+#import <GrDirectContext.h>
+#import <mtl/GrMtlBackendContext.h>
+#import <mtl/GrMtlTypes.h>
+
+#import "MetalDevice.h"
+
+extern "C"
+{
+JNIEXPORT jlong JNICALL Java_org_jetbrains_skiko_context_MetalContextHandler_makeMetalContext(
+    JNIEnv* env, jobject contextHandler, jlong devicePtr)
+{
+    @autoreleasepool {
+        MetalDevice *device = (__bridge MetalDevice *) (void*) devicePtr;
+        GrMtlBackendContext backendContext = {};
+        backendContext.fDevice.retain((__bridge GrMTLHandle) device.adapter);
+        backendContext.fQueue.retain((__bridge GrMTLHandle) device.queue);
+        return (jlong) GrDirectContext::MakeMetal(backendContext).release();
+    }
+}
+
+JNIEXPORT jlong JNICALL Java_org_jetbrains_skiko_context_MetalContextHandler_makeMetalRenderTarget(
+    JNIEnv * env, jobject contextHandler, jlong devicePtr, jint width, jint height)
+{
+    @autoreleasepool {
+        MetalDevice *device = (__bridge MetalDevice *) (void *) devicePtr;
+        GrBackendRenderTarget* renderTarget = NULL;
+
+        id<CAMetalDrawable> currentDrawable = [device.layer nextDrawable];
+        if (!currentDrawable) return 0;
+        device.drawableHandle = currentDrawable;
+        GrMtlTextureInfo info;
+        info.fTexture.retain((__bridge GrMTLHandle) currentDrawable.texture);
+        renderTarget = new GrBackendRenderTarget(width, height, 0, info);
+        return (jlong) renderTarget;
+    }
+}
+
+JNIEXPORT void JNICALL Java_org_jetbrains_skiko_context_MetalContextHandler_finishFrame(
+    JNIEnv *env, jobject contextHandler, jlong devicePtr)
+{
+    @autoreleasepool {
+        MetalDevice *device = (__bridge MetalDevice *) (void *) devicePtr;
+
+        id<CAMetalDrawable> currentDrawable = device.drawableHandle;
+
+        if (currentDrawable) {
+            id<MTLCommandBuffer> commandBuffer = [device.queue commandBuffer];
+            commandBuffer.label = @"Present";
+            [commandBuffer presentDrawable:currentDrawable];
+            [commandBuffer commit];
+            device.drawableHandle = nil;
+        }
+    }
+}
+
+} // extern C
```

Comment:
You can use `extern "C"` as part of function signature to avoid such constructions/confusions

Reply:
Looking at the nearest files, it is done the same way with extern "C" block, so I'd keep it consistent